### ISSUE TITLE
Issue with ignore or only

### ISSUE THREAD

**Issue by thaiat** [babel/babel#1693]
Hi, 

i m using `gulp` + `browserify` + `browserify-shim` + `babelify`
It looks that something change regarding the treatment of `ignore` or `only` option in `babel` between version `5.4.7` and `5.5.0`

Here is my config for the browserify transforms in `package.json`
```json
"browserify": {
        "transform": [
            "html2js-browserify",
            "jadeify",
            "cssify",
            "browserify-shim",
            "brfs", [
            "babelify", {
                    "ignore": "**/bower_components/**"
                }
            ]
        ]
    },
```

Using latest of `babelify` (6.1.2) which requires `"^5.0.0"` of `babel-core`, i get several messages like these:
```
[BABEL] Note: The code generator has deoptimised the styling of "/Users/thaiat/Documents/Phonegap/test-gen/bower_components/angular-animate/angular-animate.js" as it exceeds the max of "100KB".
[BABEL] Note: The code generator has deoptimised the styling of "/Users/thaiat/Documents/Phonegap/test-gen/bower_components/angular-ui-router/release/angular-ui-router.js" as it exceeds the max of "100KB".
[BABEL] Note: The code generator has deoptimised the styling of "/Users/thaiat/Documents/Phonegap/test-gen/bower_components/ngCordova/dist/ng-cordova.js" as it exceeds the max of "100KB".
[BABEL] Note: The code generator has deoptimised the styling of "/Users/thaiat/Documents/Phonegap/test-gen/bower_components/ionic/release/js/ionic.js" as it exceeds the max of "100KB".
```

and the bundle is not usable.

If i change the package.json of `babelify` to explictly require version `5.4.7` of babel-core, everything works.
Issue is the same with `ignore` and `only`

There are a lot of commits between 5.4.7 and 5.5.0 so i couldn't figure out where it's coming from....

**Comment by thaiat** [comment]
see also https://github.com/babel/babelify/issues/88

**Comment by Dignifiedquire** [comment]
This looks similar to the issue I'm having described in https://github.com/babel/babel-loader/issues/76
Where these messages started appearing, which I interpreted as the ignore options simply not being followed.

**Comment by thaiat** [comment]
that would suggest the bug is in babel...

**Comment by jmm** [comment]
I tried to debug this a bit and the best I could do so far was to notice that a call to `shouldIgnore()` was dropped here: https://github.com/babel/babel/commit/32f19aff99559e36c3d4fbb946af378a8126b8bf#diff-161c990af39be5f537c143be7eb7916cL498. It looks like perhaps there should be a call to `shouldIgnore()` in `generate()`?: https://github.com/babel/babel/blob/32f19aff99559e36c3d4fbb946af378a8126b8bf/src/babel/transformation/file/index.js#L607

**Comment by thaiat** [comment]
@jmm i found a fix and published a PR
let me know if it works 
mean time you can install `npm i thaiat/babel` to try it out

**Comment by sebmck** [comment]
Fixed as of 5.5.4, thanks!

**Comment by thaiat** [comment]
:+1:
